### PR TITLE
Medical - Add Vehicle Crashes setting

### LINKED ISSUE #7368: Disable vehicle collision option missing
**Is your enhancement related to a problem?**
After medical rewrite the setting to disable damage from vehicle collisions is MIA

**Solution you'd like:**
A settings to disable damage from vehicle collisions

**Additional context:**
Because ARMA is ARMA vehicle's sometimes behave, badly, and damage inflicted by this behavior is generally out of proportion/non recoverable, leaving a forced re-spawn as only option

### PR DESCRIPTION
**Pull request by Salluci**
**When merged this pull request will:**
- Close #7368.
- Adds CBA setting to disable damage to players from vehicle collision.

Translations were pulled from pre-rewrite branch.
Works in Singleplayer, hosted Multiplayer, and dedicated Multiplayer.

### CODE REVIEW

File: addons/medical_engine/functions/fnc_handleDamage.sqf [hunk 1]
```diff
@@ -211,9 +211,12 @@ if (
     // todo: no way to detect if stationary and another vehicle hits you
 ) exitWith {
     private _damageSelectionArray = [
-        HITPOINT_INDEX_HEAD, 1, HITPOINT_INDEX_BODY, 1, HITPOINT_INDEX_LARM, 1, 
+        HITPOINT_INDEX_HEAD, 1, HITPOINT_INDEX_BODY, 1, HITPOINT_INDEX_LARM, 1,
         HITPOINT_INDEX_RARM, 1, HITPOINT_INDEX_LLEG, 1, HITPOINT_INDEX_RLEG, 1
     ];
+    if !(EGVAR(medical,enableVehicleCrashes)) then {
+      _newDamage = 0;
+    };
```
Comment (mharis001) [inline addons/medical_engine/functions/fnc_handleDamage.sqf:219]: Fix indentation and move the `!` inside the parentheses.

File: addons/medical_damage/stringtable.xml [hunk 2]
```diff
@@ -697,5 +697,37 @@
             <Polish>Szansa że osoba straci przytomność gdy jej ból jest powyżej tolerowalnego progu podczas otrzymywania obrażeń.</Polish>
             <Japanese>ユニットがダメージを受けた時に痛みが許容しきい値を超えた場合、気絶をする確立を設定します。</Japanese>
         </Key>
+        <Key ID="STR_ACE_Medical_Damage_enableVehicleCrashes_DisplayName">
```
Comment (mharis001) [inline addons/medical_damage/stringtable.xml:700]: Capitalize the 'e' -> `EnableVehicleCrashes...`.

File: addons/medical_damage/stringtable.xml [hunk 3]
```diff
@@ -697,5 +697,37 @@
...
+            <Portuguese>Colisão de veículos</Portuguese>
+            <French>Accident en véhicule</French>
+            <Hungarian>Járműbalesetek</Hungarian>
+            <Italian>Schianto Veicoli</Italian>
+            <Japanese>車両の衝突</Japanese>
+            <Korean>차량 사고</Korean>
+            <Chinesesimp>载具碰撞</Chinesesimp>
+            <Chinese>載具碰撞</Chinese>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_enableVehicleCrashes_Description">
```
Comment (mharis001) [inline addons/medical_damage/stringtable.xml:716]: Capitalize the 'e' -> `EnableVehicleCrashes...`.

File: addons/medical_damage/initSettings.sqf [hunk 4]
```diff
@@ -33,3 +33,12 @@
     [0, 1, 0.1, 2, true],
     true
 ] call CBA_fnc_addSetting;
+
+[
+    QEGVAR(medical,enableVehicleCrashes),
+    "CHECKBOX",
+    [LSTRING(enableVehicleCrashes_DisplayName), LSTRING(enableVehicleCrashes_Description)],
```
Comment (mharis001) [inline addons/medical_damage/initSettings.sqf:40]: Capitalize the 'e' -> `EnableVehicleCrashes...`.

File: addons/medical_damage/stringtable.xml [hunk 5]
```diff
@@ -697,5 +697,37 @@
...
+            <English>Do units take damage from a vehicle crash?</English>
```
Comment (mharis001) [inline addons/medical_damage/stringtable.xml:717]: Setting descriptions should not be questions. We tried to get rid of those with the medical rewrite.

**Review by mharis001 — CHANGES_REQUESTED** [review]

File: addons/medical_damage/stringtable.xml (hunk 5 above)
Comment (Salluci) [reply addons/medical_damage/stringtable.xml:717]: Will adjust, but can only translate for English at that point.

File: addons/medical_engine/functions/fnc_handleDamage.sqf [hunk 6]
```diff
@@ -211,9 +211,12 @@ if (
     // todo: no way to detect if stationary and another vehicle hits you
 ) exitWith {
     private _damageSelectionArray = [
-        HITPOINT_INDEX_HEAD, 1, HITPOINT_INDEX_BODY, 1, HITPOINT_INDEX_LARM, 1, 
+        HITPOINT_INDEX_HEAD, 1, HITPOINT_INDEX_BODY, 1, HITPOINT_INDEX_LARM, 1,
         HITPOINT_INDEX_RARM, 1, HITPOINT_INDEX_LLEG, 1, HITPOINT_INDEX_RLEG, 1
     ];
+    if (!EGVAR(medical,enableVehicleCrashes)) then {
+        _newDamage = 0;
```
Comment (mharis001) [inline addons/medical_engine/functions/fnc_handleDamage.sqf:218]: This still triggers the `QEGVAR(medical,woundReceived)` event. When vehicle crashes are disabled, I think we wouldn't want "vehiclecrash" wound received events.

Comment (Salluci) [reply addons/medical_engine/functions/fnc_handleDamage.sqf:218]: Fair enough. Forgot to account for that.

File: addons/medical_engine/functions/fnc_handleDamage.sqf [hunk 7]
```diff
@@ -200,6 +200,11 @@ if (
     0
 };
 
+// No need to go any further if vehicle crash damage is disabled
+if (!EGVAR(medical,enableVehicleCrashes)) exitWith {
```
Comment (mharis001) [inline addons/medical_engine/functions/fnc_handleDamage.sqf:204]: Would prefer if this condition was just part of the vehicle crash `if`'s condition.

Comment (Salluci) [reply addons/medical_engine/functions/fnc_handleDamage.sqf:204]: I need to start drinking more coffee when doing these.

File: addons/medical_damage/stringtable.xml [hunk 8]
```diff
@@ -697,5 +697,11 @@
             <Polish>Szansa że osoba straci przytomność gdy jej ból jest powyżej tolerowalnego progu podczas otrzymywania obrażeń.</Polish>
             <Japanese>ユニットがダメージを受けた時に痛みが許容しきい値を超えた場合、気絶をする確立を設定します。</Japanese>
         </Key>
+        <Key ID="STR_ACE_Medical_Damage_EnableVehicleCrashes_DisplayName">
+            <English>Enable Vehicle Crash Damage</English>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_EnableVehicleCrashes_Description">
+            <English>Controls whether passengers receive damage from vehicle collisions.</English>
```
Comment (jonpas) [inline addons/medical_damage/stringtable.xml:704]:
```suggestion
            <English>Controls whether crew receives damage from vehicle collisions.</English>
```
"Passengers" usually excludes drivers and other roles.

**Review by jonpas — APPROVED** [review]

File: addons/medical_damage/initSettings.sqf [hunk 9]
```diff
@@ -42,3 +42,12 @@
...
 ] call CBA_fnc_addSetting;
+
+[
+    QEGVAR(medical,enableVehicleCrashes),
+    "CHECKBOX",
+    [LSTRING(EnableVehicleCrashes_DisplayName), LSTRING(EnableVehicleCrashes_Description)],
+    ELSTRING(medical,Category),
+    true,
+    true
+] call CBA_fnc_addSetting;
```
Comment (jonpas) [inline addons/medical_damage/initSettings.sqf:53]:
```suggestion
] call CBA_fnc_addSetting;

```

**Review by jonpas — CHANGES_REQUESTED** [review]

**Review by jonpas — APPROVED** [review]

**Review by mharis001 — APPROVED** [review]

**Comment by kymckay** [comment]
Just reviewing some of the items from the draft changelog. I don't believe this PR should have been merged as is, medical engine cannot depend on a setting from medical damage due to the dependency tree of those components.

Either medical engine must now depend on medical damage (creating a loop in the dependency graph) or this setting should be moved to medical engine where it takes effect.

Fwiw, there was some reason this setting was removed in the rewrite, but I forget what it was. I think it may be because there's no way to consistently detect all vehicle crashes (as reflected in the code comments).

**Pull request merged**